wordpress-mobile/WordPress-iOS
# Media Library: Allow batch deletion of failed media

This PR allows failed media uploads to be deleted in 'edit' mode in the media library. Previously, you had to tap each item in turn and choose "Cancel Upload" to remove it.

**To test:**

* In the media library, start a number of items uploading
* Cause the uploads to fail (for example use airplane mode)
* Tap Edit in the navigation bar, and check that you can select the failed items
* Select some or all of the failed items and tap the trash can to delete them
* Ensure they are removed
* Ensure additional uploads complete successfully

In making this change, I also switched the media library over to do its deletions via the media coordinator, which means it no longer touches the MediaService directly.
